tests/test_ugrid.cpp

```diff
@@ -814,6 +814,306 @@ TEST( MeshUgridTest, flow_3d )
+  MDAL_MeshH m = MDAL_LoadMesh( tmpFile.c_str() );
+  ASSERT_NE( m, nullptr );
+  MDAL_Status s = MDAL_LastStatus();
+  EXPECT_EQ( MDAL_Status::None, s );
+
+  int groupCount = MDAL_M_datasetGroupCount( m );
+  ASSERT_EQ( 10, groupCount );
+
+  int faceCount = MDAL_M_faceCount( m );
+  int vertexCount = MDAL_M_vertexCount( m );
+
+  MDAL_DatasetGroupH existingGroup = MDAL_M_datasetGroup( m, 7 );
+  int datasetCount = MDAL_G_datasetCount( existingGroup );
+  std::vector<double> timeSteps( static_cast<size_t>( datasetCount ) );
+  for ( size_t i = 0; i < timeSteps.size(); ++i )
+    timeSteps[i] = MDAL_D_time( MDAL_G_dataset( existingGroup, i ) );
+  std::string refTime( MDAL_G_referenceTime( existingGroup ) );
+
+  bool success = false;
+
+  createNewDatasetGroupOnExistingFile( m, tmpFile, refTime, timeSteps, "new group scalar faces", MDAL_DataLocation::DataOnFaces, faceCount, true, 1.23, success );
+  EXPECT_TRUE( success );
+  createNewDatasetGroupOnExistingFile( m, tmpFile, refTime, timeSteps, "new group scalar vertices", MDAL_DataLocation::DataOnVertices, vertexCount, true, 4.56, success );
+  EXPECT_TRUE( success );
+
+  createNewDatasetGroupOnExistingFile( m, tmpFile, refTime, timeSteps, "new group vector faces", MDAL_DataLocation::DataOnFaces, faceCount, false, 1.23, success );
+  EXPECT_TRUE( success );
+  createNewDatasetGroupOnExistingFile( m, tmpFile, refTime, timeSteps, "new group vector vertices", MDAL_DataLocation::DataOnVertices, vertexCount, false, 4.56, success );
+  EXPECT_TRUE( success );
+
+  std::vector<double> badTimeSteps = timeSteps;
+  badTimeSteps[0] = 1234;
+  createNewDatasetGroupOnExistingFile( m, tmpFile, refTime, badTimeSteps, "new group scalar faces bad time step", MDAL_DataLocation::DataOnFaces, faceCount, true, 1.23, success );
+  EXPECT_FALSE( success );
+  badTimeSteps.erase( badTimeSteps.begin() );
+  createNewDatasetGroupOnExistingFile( m, tmpFile, refTime, badTimeSteps, "new group scalar vertices", MDAL_DataLocation::DataOnVertices, vertexCount, true, 4.56, success );
+  EXPECT_FALSE( success );
+
+  //not supported
+  createNewDatasetGroupOnExistingFile( m, tmpFile, refTime, timeSteps, "new group scalar volume", MDAL_DataLocation::DataOnVolumes, 10, true, 1.23, success );
+  EXPECT_FALSE( success );
+  //not supported
+  createNewDatasetGroupOnExistingFile( m, tmpFile, refTime, timeSteps, "new group scalar edge", MDAL_DataLocation::DataOnEdges, 10, true, 1.23, success );
+  EXPECT_FALSE( success );
+  //not supported
+  createNewDatasetGroupOnExistingFile( m, tmpFile, refTime, timeSteps, "new group scalar unknow location", MDAL_DataLocation::DataInvalidLocation, 10, true, 1.23, success );
+  EXPECT_FALSE( success );
+
+  MDAL_CloseMesh( m );
+
+  m = MDAL_LoadMesh( tmpFile.c_str() );
+  ASSERT_NE( m, nullptr );
+  s = MDAL_LastStatus();
+  EXPECT_EQ( MDAL_Status::None, s );
+
+  groupCount = MDAL_M_datasetGroupCount( m );
+  ASSERT_EQ( 14, groupCount );
+
```

Comment:
would be good to check at least one scalar, one vector value on various data locations

Reply:
dataset values are checked in `createNewDatasetGroupOnExistingFile()`